File: 050_powx_n/Solution.java
class Solution {
    public double quickMul(double x, int n) {
        if (n == 0) return 1.0;
        double y = quickMul(x, n / 2);
        return n % 2 == 0 ? y * y : y * y * x;
    }

    public double myPow(double x, int n) {
        return n >= 0 ? quickMul(x, n) : 1.0 / quickMul(x, -n);
    }

    public static void main(String[] args) {
        System.out.println(new Solution().myPow(2, 10));
    }
}

File: 050_powx_n/Solution.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        def quick_mul(x: float, n: int):
            if n == 0: return 1
            y = quick_mul(x, n // 2)
            return y * y if n % 2 == 0 else y * y * x

        return quick_mul(x, n) if n >= 0 else 1 / quick_mul(x, -n)


if __name__ == '__main__':
    solution = Solution()
    print(solution.myPow(2, 10))

